apps/addons/buttons.py

```diff
@@ -233,11 +234,13 @@ def __init__(self, text, url, os=None, file=None):
 
 # Cache it for a year.
 @cache_page(60 * 60 * 24 * 365)
+@non_atomic_requests
```

Comment:
I wonder if you need the `non_atomic_requests` decorator on top instead. It would make sense to me at least...